programs/governance-rewards/src/instructions/claim.rs

```diff
@@ -0,0 +1,163 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    associated_token::get_associated_token_address,
+    token::{self, Token, TokenAccount},
+};
+use std::mem::size_of;
+
+use crate::{
+    distribution_payout_seeds,
+    error::GovernanceRewardsError,
+    state::{
+        claim_data::ClaimData,
+        distribution::Distribution,
+        preferences::{ResolutionPreference, UserPreferences},
+    },
+};
+
+#[derive(Accounts)]
+pub struct Claim<'info> {
+    #[account(mut)]
+    distribution: Account<'info, Distribution>,
+
+    #[account(
+        mut,
+        seeds = [distribution.key().as_ref(), b"user data".as_ref(), claimant.key().as_ref()],
+        bump
+    )]
+    user_data: Account<'info, ClaimData>,
+
+    #[account(
+        mut,
+        address = user_data.chosen_option(&distribution).wallet
+    )]
+    rewards_account: Account<'info, TokenAccount>,
+
+    /// CHECK: Not read
+    #[account(seeds = [b"payout authority".as_ref(), distribution.key().as_ref()], bump)]
+    payout_authority: AccountInfo<'info>,
+
+    #[account(init_if_needed, payer = caller, space = size_of::<TokenAccount>())]
+    to_account: Account<'info, TokenAccount>,
```

Comment:
The `init` case will probably not currently work. I don't think the TokenAccount::owner would be set?

Reply:
I guess we don't have to support the init case. I was under the impression that the TokenAccount would get initialized before the rest of the program runs. But I'll write a test to find out.

Reply:
(No longer initializes for you)